tests/test_java.py

```diff
@@ -691,3 +691,19 @@ def test_meminfo_logged(
     ) as profiler:
         snapshot_pid_profile(profiler, application_pid)
         assert "async-profiler memory usage (in bytes)" in caplog.text
+
+
+@pytest.mark.parametrize("in_container", [True])
+def test_java_frames_include_no_semicolons(
+    tmp_path: Path,
+    application_pid: int,
+) -> None:
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        collapsed = snapshot_pid_profile(profiler, application_pid).stacks
+        assert is_function_in_collapsed(".main([Ljava/lang/String|)", collapsed)
+        assert not is_function_in_collapsed(";_[j];", collapsed)
```

Comment:
What is checked here? Why will we ever have an empty function name? :sweat_smile:

Reply:
It's testing for a object-kind return type of a function.
I've run this test against a non-patched version of async-profiler and this pattern appears in collapsed stacls.

Reply:
Ahh because if any function returns an object, it'll be `func()Ljava/lang/String;_[j];`. I see.